### DISCUSSION TITLE
IaC / Terraform files - location recommendations

### DISCUSSION THREAD

**Discussion by saifmode** [DavidVujic/python-polylith#300] [General]
What location in a polylith would you recommend for keeping Terraform / IaC files? I'm thinking they would go in project folders, but interested to hear opinions.

**Comment by DavidVujic** [comment]
We've added things like that in a `deploy` folder at the root (using AWS CDK) before. For K8s configs, I have a blog post with example code (very experimental though) with an attempt to add configs in a similar way as with Polylith bricks: https://davidvujic.blogspot.com/2024/08/simple-kubernetes-in-python-monorepos.html

For terraform, I have not done anything else than have it all in a `terraform` folder (not a Polylith repo). But having that in the project folders makes a lot of sense to me!

**Comment by ybenitezf** [comment]
I have an example with [aws CDK](https://dev.to/ybenitezf/aws-cdk-app-with-polylith-code-architecture-30e3). In [this case](https://github.com/ybenitezf/cdk_polylith/) the IaC is spread over the bricks and the cdk app that do the actual deployment is in the projects folders.

Take into account that terraform has a [Cloud Development Kit](https://developer.hashicorp.com/terraform/cdktf) that support Python too. So you can include your Infra as part of the Python code.

But for the case of the traditional `.tf` fles, i think they make more sense in the projects folder - if the project is simple.  At the end the [propourse of the project in the polylith architecture](https://polylith.gitbook.io/polylith/architecture/2.6.-project) is to build the artifacts (`.whl` or `.tar.gz`) which we deploy in our test, staging and production environments. If your architecture is complex and is not written in the same language then use a separated repository only for the IaC, and have some way to use the artifacts produced by the polylith repository.
